### ISSUE TITLE
Optimal dataset size?

### ISSUE THREAD

**Issue by kodlan** [thtrieu/darkflow#826]
What is an optimal dataset size for training (images per class)?
I understand it depends, but at least some number that should work will be enough.
What loss are you achieving with such a dataset and how many steps its required?

**Comment by jatinmandav** [comment]
I don't really know about optimal dataset size, but once I used about 1000 (+- 80) images each for 10 different classes of traffic signs, and was able to detect them. I trained the network for 40 - 50 epochs. I guess if you have less number of classes, you can get away with 500-600 images with 50 epochs! Let me know how this worked out for you!

**Comment by kodlan** [comment]
Thanks for your answer.
How do you decide when it is time to stop training?
When I tried, loss stops improving way before 40-50 epochs. Do you just continue to train, even when loss is not improving?
(I was using pre-trained weights to train on my dataset)

**Comment by jatinmandav** [comment]
The best time to stop training should be when your loss reaches around 1 (or 0.1).
Does your network work good for you when your loss stops changing?

**Comment by kodlan** [comment]
Well it detects something, but confidence is pretty low, like 0.1-0.2.
But I think it's the problem with my dataset, as it has like ~500 images.
I'll try getting some additional images.

**Comment by jatinmandav** [comment]
Okay! Let me how it turns out! And if it resolves, don't forget to close the issue!
Regards
Mandav
